src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionTestResultConsentViewController.swift

```diff
@@ -27,10 +27,14 @@ class ExposureSubmissionTestResultConsentViewController: DynamicTableViewControl
 	override func viewDidLoad() {
 		super.viewDidLoad()
 		self.title = AppStrings.AutomaticSharingConsent.consentTitle
-		self.navigationController?.navigationItem.rightBarButtonItem = nil
 		setupView()
 	}
 	
+	override func viewDidAppear(_ animated: Bool) {
+		super.viewDidAppear(animated)
+		self.navigationItem.rightBarButtonItem = nil
```

Comment:
Why is this ADDING a cancel button??? 🤯

Reply:
Instead of removing it for every following view controller it now just removes it from this single view controller. This code is not in the view controller where the button was missing 😉

Reply:
Yes exactly as Nick said, we pass the nil to the Navigation Controller which effect some other VC that's why this viewDidAppear makes nil only in TestResultConsentVC. Therefore that cancel button missing because of this mess.

Reply:
> Instead of removing it for every following view controller it now just removes it from this single view controller. This code is not in the view controller where the button was missing 😉

Thanks Nick for the quick reply.

Reply:
And all because people can't dismiss modal views properly… Thanks guys!